File: Ruby/P2/Irrgarten/Weapon.rb
#encoding:utf-8

=begin

    Autor: Isabel Morro Tabares 

    Armas que utiliza el jugador en los ataques durante los combates

=end

require_relative 'Dice'

module Irrgarten
    class Weapon

        # Constructor

        def initialize (p, u)
            @power = p
            @uses = u 
            @dice = Dice.new    
        end

        # Métodos públicos

        def attack 
            resultado = 0
            if @uses > 0 
                @uses = @uses - 1
                resultado = @power
            end 
            resultado
        end

        # Implementa la decisión de si un arma es descartada
        def discard
            @dice.discardElement(@uses)
        end

        # To string
        def to_s
            cadena = "W[" + @power.to_s() + ", " + @uses.to_s() + "]"
        end
    end

end

File: Java/P1/irrgaten/src/irrgarten/Weapon.java

package irrgarten;

/**
 * @brief   Armas que utiliza el jugador en los ataques durante los combates
 * @author  Isabel Morro Tabares
 */
public class Weapon {

    // Variables privadas
    
    private float power;
    private int uses;
    Dice dice; //???????????????
    
    // Constructor
    
    public Weapon (float _power, int _uses){
        
        power = _power;
        uses = _uses;
        dice = new Dice();
    }
    
    // Métodos públicos

    public float attack (){
        
        float result = 0;
        
        if (uses > 0){
            uses--;
            result = power;
        }
        
        return result;
    }
    
    /**
     * @brief   Toma decisión de si un arma es descacartada
     * @return  true si descartado o false en caso contrario
     */
    public boolean discard(){
        return dice.discardElement(uses);
    }
    
    // To string
    @Override
    public String toString(){
        String cadena = "W[" + Float.toString(power) + ", " + 
                        Integer.toString(uses) + "]";
        return cadena;
    }
}
